snapcraft/internal/errors.py

```diff
@@ -202,17 +202,39 @@ def __init__(self, remote):
         super().__init__(remote=remote)
 
 
-class InvalidDesktopFileError(SnapcraftError):
+class MissingDesktopFileError(SnapcraftError):
 
     fmt = (
         'Failed to generate desktop file: '
-        'Invalid desktop file {filename!r}: {message}.'
-        # FIXME include how to fix each of the possible desktop file errors.
-        # https://bugs.launchpad.net/snapcraft/+bug/1727435
-        # --elopio - 2017-10-25
+        'The file {filename!r} defined in the app {app_name!r} does not exist.'
+        
+    )
+
+    def __init__(self, filename, app_name):
+        super().__init__(filename=filename, app_name=app_name)
+
+
+class MissingDesktopSectionError(SnapcraftError):
+
+    fmt = (
+        'Failed to find any Desktop Entry: '
```

Comment:
I suggest keeping this first line "Failed to generate desktop file" as above. All our error messages should ideally contain three things:

1. What happened
2. Why did it happen
3. How to fix it

(3) is a little harder here, but we should be able to nail (1) and (2). (1) is still "Failed to generate desktop file", and I suspect it's the same for all three of these error classes.